param/parameterized.py

```diff
@@ -3358,6 +3375,11 @@ def __param_inheritance(mcs, param_name, param):
                 else:
                     setattr(param, slot, default_val)
 
+            # Avoid crosstalk between mutable slot values in different Parameter objects
+            v = getattr(param, slot)
+            if (not getattr(param, "constant", False) and is_mutable(v)):
+                setattr(param, slot, copy.copy(v))
```

Comment:
In a simple case like below do we need to make a shallow copy of the objects? Currently it does.

```python
class P(param.Parameterized):
    s = param.Selector(objects=[1, 2]))

Reply:
Yes, if `per_instance` is True, I think we should be making a copy of the objects list per instance.  I think we should not  make any copies if `per_instance=False`.